openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -26,6 +26,7 @@ class LogoutView(TemplateView):
 
     # Keep track of the page to which the user should ultimately be redirected.
     default_target = reverse_lazy('cas-logout') if settings.FEATURES.get('AUTH_USE_CAS') else '/'
+    user_have_saml_provider = False
```

Comment:
you could call is_saml_user inside allow_saml_logout_from_url since you are using this variable once, obviously you should add the user  as parameter to allow_saml_logout_from_url and it should be user_has....

Reply:
The problem here is, if I call it after the logout operation on line 61, the request.user will return an AnonymousUser instance instead of original user. @andrey-canon 
Grammatical error, changed...